cassandra/pool.py

```diff
@@ -572,14 +573,16 @@ def return_connection(self, connection):
             in_flight = connection.in_flight
 
         if connection.is_defunct or connection.is_closed:
```

Comment:
Seems like a lot of the same code as in `HostConnection` does it make sense to merge any of this into a `Connection` method?